### PR TITLE
sonic yang changes for BUM storm control

### PR DESCRIPTION
**Pull request by mohan-selvaraj**
#### Why I did it
Sonic yang model for BUM storm control

#### How I did it
Added yang model and the corresponding test cases. 

#### How to verify it
yang model test case for storm control

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

#### Description for the changelog

Addition of storm control yang model and corresponding tests to verify the same.

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-storm-control.yang [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        "SONiC";
+
+    description
+        "This module defines configuration and operational state
+        data for broadcast, unknown-unicast and unknown-multicast (BUM) storm-control.  
+        storm-control can be enabled on ethernet interfaces. 
+        Rate is defined in bits-per-second. All three types of storm-control can be 
+        enabled on an interface.";
+
+    revision 2019-09-03 {
```
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-storm-control.yang:24]: Update the revision date.

File: src/sonic-yang-models/yang-models/sonic-storm-control.yang [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        description 
+            "Initial revision";
+    }
+
+    container sonic-storm-control {
+        container PORT_STORM_CONTROL {
+            list PORT_STORM_CONTROL_LIST {
+                key "ifname storm_type";
+                /*scommon:key-pattern "PORT_STORM_CONTROL|{ifname}:{storm_type}";*/
```
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-storm-control.yang:33]: Why do we have ":" in the key pattern, we should use "|"

File: src/sonic-yang-models/tests/files/sample_config_db.json [hunk 3]
```diff
@@ -694,6 +694,17 @@
...
+        "PORT_STORM_CONTROL": {
+            "Ethernet0|broadcast": {
+                "kbps": "10000"
+            },
+            "Ethernet0|unknown-multicast": {
+                "kbps": "30000"
+            },
+            "Ethernet0|unknown-unicast": {
+                "kbps": "20000"
+            }
```
Comment (venkatmahalingam) [inline src/sonic-yang-models/tests/files/sample_config_db.json:707]: Please write the actual UT cases as well e.g add cases in src/sonic-yang-models/tests/yang_model_tests/tests for this feature.

File: src/sonic-yang-models/yang-models/sonic-storm-control.yang [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+                    }
+                }
+                leaf storm_type {
+                    type enumeration{
+                        enum broadcast;
+                        enum unknown-unicast;
+                        enum unknown-multicast;
+                    }
+                }
+                leaf kbps {
```
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-storm-control.yang:46]: we can we use rate instead of kbps field name?

**Comment by zhangyanzhao** [comment]
@mohan-selvaraj  could you please join the YANG subgroup meeting and review with team? Thanks.

**Comment by zhangyanzhao** [comment]
@mohan-selvaraj  please help to fix the conflict asap.

File: src/sonic-yang-models/yang-models/sonic-storm-control.yang (hunk 1 above)
Comment (mohan-selvaraj) [reply src/sonic-yang-models/yang-models/sonic-storm-control.yang:24]: Done

File: src/sonic-yang-models/yang-models/sonic-storm-control.yang (hunk 2 above)
Comment (mohan-selvaraj) [reply src/sonic-yang-models/yang-models/sonic-storm-control.yang:33]: Updated

File: src/sonic-yang-models/yang-models/sonic-storm-control.yang (hunk 4 above)
Comment (mohan-selvaraj) [reply src/sonic-yang-models/yang-models/sonic-storm-control.yang:46]: rate might indicate the % of linerate. kbps would be a constant value irrespective of line-rate. Hence chose kbps.

File: src/sonic-yang-models/tests/files/sample_config_db.json (hunk 3 above)
Comment (mohan-selvaraj) [reply src/sonic-yang-models/tests/files/sample_config_db.json:990]: Done

File: src/sonic-yang-models/yang-models/sonic-storm-control.yang (hunk 4 above)
Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-storm-control.yang:46]: Sorry...I meant meaningful name than kbps e.g storm_limit..

Comment (mohan-selvaraj) [reply src/sonic-yang-models/yang-models/sonic-storm-control.yang:46]: Okay. Do you recommend changing this in all the instances (CLI and orchagent) or only here.

Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-storm-control.yang:46]: I recommend changing it in all places, but that can be done in a separate PR.

**Review by venkatmahalingam — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**